src/redis-benchmark.c

```diff
@@ -1934,6 +1934,14 @@ int main(int argc, char **argv) {
             free(cmd);
         }
 
+        if (test_is_selected("xadd")) {
+            char streamId[64];
+            sprintf(streamId, "mystream-%d-%d", (int)time(NULL), rand());
+            len = redisFormatCommand(&cmd,"XADD %s * %s %s", streamId, tag, data);
```

Comment:
Tag is used for contolling which cluster shard the key belongs to. It needs to be part of the key.

Why include a timestamp and random number in the key name? For the list tests, the key is just mylist:{tag}.

I don't think the stream test should be located in the middle of the list tests (between rpush and lpop).

Reply:
moved it down and located it at the bottom. what about something like this
```
        if (test_is_selected("xadd")) {
            len = redisFormatCommand(&cmd,"XADD mystream%s * %s %s", tag, tag, data);
            benchmark("XADD",cmd,len);
            free(cmd);
        } 

  ```

this confirms to the notation...

Reply:
Yes, that's better. Key name includes tag. :+1:

But the tag is also used as field name. Why is that? When redis-benchmark runs without cluster mode, tag is "" and with cluster mode it is something like ":{tag}" (where "tag" is replaced by something else later) so the stream content has different sizes depending on cluster mode on/off. Isn't it better to use a fixed field name like "myfield" instead? (`XADD mystream:{tag} * myfield data`)

Reply:
@zuiderkwast , made changes as suggested - let me know if you need anything else. Thanks.